src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -103,20 +103,27 @@ public static class ActivatorUtilities
                         {
                             ThrowMarkedCtorDoesNotTakeAllProvidedArguments();
                         }
-                    }
 
-                    if (isPreferred || bestLength < length)
-                    {
+                        seenPreferred = true;
                         bestLength = length;
                         bestMatcher = matcher;
                         multipleBestLengthFound = false;
                     }
-                    else if (bestLength == length)
+                    else if (!seenPreferred)
                     {
-                        multipleBestLengthFound = true;
+                        if (bestLength < length)
+                        {
+                            // When a preferred constructor is found, it is always selected however subsequent constructors
+                            // will be selected if they have more parameters.
```

Comment:
I don't think this comment is true any more.

Reply:
Oops thanks